spec/features/publish_pact_all_in_one_spec.rb

```diff
@@ -0,0 +1,38 @@
+RSpec.describe "publishing a pact using the all in one endpoint" do
+  # TODO not sure about "role"
+  # TODO validation
+  # TODO merge tags
+  # TODO merge branches
+  # TODO merge pacts
+  # TODO warn when pact is overwritten with different content
+  let(:request_body_hash) do
+    {
+      :pacticipantName => "Foo",
+      :versionNumber => "1",
+      :tags => ["a", "b"],
+      :branch => "main",
+      :buildUrl => "http://ci/builds/1234",
+      :contracts => [
+        {
+          :role => "consumer",
```

Comment:
Not sure about the field "role". We need some way of saying "is this a consumer defined contract or a provider defined contract". "Type" is such an overloaded, non-descriptive term. "contractAuthor" sounds like it's a person. "contractSpecifier" also. "contractSource"?

Reply:
@mefellows @TimothyJones thoughts?

Reply:
role made sense to me, it was clear was what it was when I saw the value for it